### PR TITLE
Use `doc`, not `test` for fn item inner attr example

### PR DESCRIPTION
**Pull request by Havvy**
Using `test` causes a deny-by-default lint to trigger. See
https://github.com/rust-lang/rust/pull/79003 for more information.

### CODE REVIEW

**Comment by ehuss** [comment]
Thanks!  The sentence above the example should probably be updated (`The function will only be available while running tests.`) and the function name (`test_only`) should also probably be changed.

I think it would also be a good idea to document this behavior somehow.  AFAIK ([rustc_builtin_macros](https://github.com/rust-lang/rust/blob/40624dde6cdfad2576b4d795c97583188a814884/compiler/rustc_builtin_macros/src/lib.rs#L89-L95)), the only attributes this affects is `test` and `bench`.  I don't know how soon https://github.com/rust-lang/rust/issues/54726 might get stabilized, but if it might be a long time, maybe a brief mention at https://github.com/rust-lang/reference/blob/master/src/attributes/testing.md#the-test-attribute that says something along the lines of:  "similar to proc-macros, the `#[test]` attribute only works as an [outer attribute], and cannot be used as an [inner attribute]".

File: src/items/functions.md [hunk 1]
```diff
@@ -295,7 +295,7 @@ available while running tests.
 
 ```rust
 fn test_only() {
-    #![test]
+    #![doc("Example")]
```
Comment (petrochenkov) [inline src/items/functions.md:298]:
```suggestion
    #![doc = "Example"]
```
`doc("Example")` is not rejected by validation, but it doesn't do anything.

**Comment by Havvy** [comment]
I want to focus this PR on just getting toolstate working, so the update to the test file shall be in its own PR.

**Review by ehuss — APPROVED** [review]
Thanks! Seems reasonable, I don't think there is any rush.  It's an extreme edge case, that hopefully some day will be removed, in which case there will be no need to document it.
